src/actions/index.ts

```diff
@@ -0,0 +1,50 @@
+export enum ActionKeys {
+    LOAD = "LOAD",
+    LOADED = "LOADED",
+    TOUCH = "TOUCH",
+    TOUCHED = "TOUCHED",
+}
+
+export interface LoadAction {
+    type: ActionKeys.LOAD;
+    id: string;
+}
+
+export const load = (id: string): LoadAction => ({
+   type: ActionKeys.LOAD,
+   id,
+});
+
+export interface LoadedAction {
+    type: ActionKeys.LOADED;
+    oppName: string;
+    touches: number;
+}
+
+export const loaded = (oppName: string, touches: number): LoadedAction => ({
+    type: ActionKeys.LOADED,
+    oppName,
+    touches,
+});
+
+export interface TouchAction {
+    type: ActionKeys.TOUCH;
+    id: string;
```

Comment:
don't think you need the ID here

Reply:
not yet, put it in there so I would be able to do the update call without needing the full state. assuming that happens in the action creator i'd need that right?